lib/trip_dispatcher.rb

```diff
@@ -75,19 +75,93 @@ def load_trips
           id: raw_trip[:id].to_i,
           driver: driver,
           passenger: passenger,
-          start_time: raw_trip[:start_time],
-          end_time: raw_trip[:end_time],
+          start_time: Time.parse(raw_trip[:start_time]),
+          end_time: Time.parse(raw_trip[:end_time]),
           cost: raw_trip[:cost].to_f,
           rating: raw_trip[:rating].to_i
         }
 
         trip = Trip.new(parsed_trip)
+
+        # set up relations
         driver.add_trip(trip)
         passenger.add_trip(trip)
         trips << trip
       end
 
-      trips
+      return trips
+    end
+
+    # helper method to calculate the next new id for the request_trip method
+    def gets_new_trip_id
+      next_id = @trips.max_by {|trip| trip.id}
+      return next_id.id
+    end
+
+    # helper method to get next driver with status available and oldest recent trip.
+    def driver_selection
+      drivers_available = drivers.find_all do |driver|
+        driver.status == :AVAILABLE
+      end
```

Comment:
Good use of a helper method here! This makes the code that calls this method much easier to read.